src/test/scala/com/bwsw/cloudstack/vault/server/cloudstack/CloudStackServiceTestSuite.scala

```
@@ -2,23 +2,24 @@ package com.bwsw.cloudstack.vault.server.cloudstack
 
 import java.util.UUID
 
-import com.bwsw.cloudstack.vault.server.BaseSuite
+import com.bwsw.cloudstack.vault.server.MockConfig._
+import com.bwsw.cloudstack.vault.server.BaseTestSuite
 import com.bwsw.cloudstack.vault.server.cloudstack.entities.{Command, Tag}
-import com.bwsw.cloudstack.vault.server.cloudstack.util.ApacheCloudStackTaskCreator
+import com.bwsw.cloudstack.vault.server.cloudstack.util.CloudStackTaskCreator
 import com.bwsw.cloudstack.vault.server.cloudstack.util.exception.CloudStackCriticalException
 import org.scalatest.FlatSpec
 
 /**
   * Created by medvedev_vv on 25.08.17.
   */
-class CloudStackServiceTestSuite extends FlatSpec with TestData with BaseSuite {
+class CloudStackServiceTestSuite extends FlatSpec with TestData with BaseTestSuite {
```

What the case the negative tests check (it's about tests naming)? You stub a method that throws an exception. So actually you want to check that a tested method invokes a stub method and doesn't swallow an exception of specific type but a test name tell us that it just should throw an exception. Make the names more clearly. And it would be better to add a test case when a method swallows an exception (that has got a type that is defferent from CloudStackCriticalException) - Mockito can help you or other library. Also there are no tests of setResourceTag method and the cases when there hasn't been found something by ID (getOrElse statement in each method of CloudStackService)

---

ok